---
title: "Lab_13_Jay_English"
author: "Jay English"
date: "2024-04-10"
output: 
  html_document:
    toc: true
    toc_float: true
    number_sections: true
    theme: spacelab
    highlight: breezedark
---

```{r}
library(tidyverse)
library(ggtree)
library(treeio)
library(ggimage)
library(rphylopic)
library(ggeasy)
```

# Basic Trees

```{r}
# Import tree data. Since it's a newick file, we'll use read.tree.

tree <- read.tree("data/tree_newick.nwk")

# build a ggplot with geom_tree

ggplot(tree) + geom_tree() + theme_tree()

# Or shorthand
ggtree(tree)
```

The treescale geom adds a scale bar, or you could change the theme to theme_tree2. The horizontal axis shows the amount of genetic change while the branches represent divergences in taxa. Units for branch lengths are usually nucleotides substitutions per site. or changes per 100 bases.

```{r}
# add a scale
ggtree(tree) + geom_treescale()

# or add the entire scale to the x axis with theme_tree2()
ggtree(tree) + theme_tree2()
```

To disable scaling and produce a cladogram, set branch.length="none" within ggtree.

```{r}
ggtree(tree, branch.length="none")
```

You can also change other options like size, color, and line type.

```{r}
ggtree(tree, branch.length="none", color="blue", size=2, linetype=3)
```

# Other Tree Geoms

Plot objects can store the basic layouts of a plot, and add more layers as we desire.

## Add nodes and tip points

```{r}
# create the basic plot
p <- ggtree(tree)
# add node points
p + geom_nodepoint()
# add tip points
p + geom_tippoint()
# Label the tips
p + geom_tiplab()
```

# Tree Annotation

## Internal Node Number

```{r}
ggtree(tree) + geom_text(aes(label=node), hjust=-.3)

```

Or you can just use MRCA to pull a specific node number:

```{r}

ggtree(tree) + geom_tiplab() 

MRCA(tree, c("C", "E"))

MRCA(tree, c("G", "H"))
```

## Labeling Clades

You can use geom_cladelabel to add a layer annotating a selected clade 

```{r}
ggtree(tree) + 
  geom_cladelabel(node=17, label="Some random clade", color="red")
```

You can still add tip labels

```{r}
ggtree(tree) + 
  geom_tiplab() + 
  geom_cladelabel(node=17, label="Some random clade", 
                  color="red2", offset=.8)
```

```{r}
ggtree(tree) + 
  geom_tiplab() + 
  geom_cladelabel(node=17, label="Some random clade", 
                  color="red2", offset=.8) + 
  geom_cladelabel(node=21, label="A different clade", 
                  color="blue", offset=.8)
```

However, simply adding them is not enough. We'll need to adjust them so they look better.

```{r}
ggtree(tree) + 
  geom_tiplab() + 
  geom_cladelabel(node=17, label="Some random clade", 
                  color="red2", offset=.8, align=TRUE) + 
  geom_cladelabel(node=21, label="A different clade", 
                  color="blue", offset=.8, align=TRUE) + 
  theme_tree2() + 
  xlim(0, 70) + 
  theme_tree()
```
We could also just highlight these clades

```{r}
ggtree(tree) + 
  geom_tiplab() + 
  geom_hilight(node=17, fill="gold") + 
  geom_hilight(node=21, fill="purple")
```

# Connecting Taxa

Evolutionary eventa can be visualized with some simple annotations using geom_taxalink. This layer draws lines between two nodes.

```{r}
ggtree(tree) + 
  geom_tiplab() + 
  geom_taxalink("E", "H", color="blue3") +
  geom_taxalink("C", "G", color="orange2", curvature=-.9)
```

# Advanced Tree Annotation

```{r}
# Read the data
tree2 <- read.beast("data/flu_tree_beast.tree")
# supply a most recent sampling date so you get the dates
# and add a scale bar
ggtree(tree2, mrsd="2013-01-01") + 
  theme_tree2() 
# Finally, add tip labels and adjust axis
ggtree(tree2, mrsd="2013-01-01") + 
  theme_tree2() + 
  geom_tiplab(align=TRUE, linesize=.5) + 
  xlim(1990, 2020)
```

msaplot can take a tree object and the path to a FASTA alignment. Pass + coord_polar(theta = "y") to the end of the command:

```{r}
msaplot(p=ggtree(tree2), fasta="data/flu_aasequence.fasta", window=c(150, 175))
```


## Many Trees

ggtree will let you plot many trees at once, and you can facet them the normal way in ggplot2. Let's generate 3 replicates each of 4 random trees with 10, 25, 50, and 100 tips, plotting them all.

```{r}
set.seed(42)
trees <- lapply(rep(c(10, 25, 50, 100), 3), rtree)
class(trees) <- "multiPhylo"
ggtree(trees) + facet_wrap(~.id, scale="free", ncol=4) + ggtitle("Many trees. Such phylogenetics. Wow.")
```

## Plot trees with other data

```{r}
# Generate a random tree with 30 tips
tree2 <- rtree(30)
# Make the original plot
p <- ggtree(tree2)
# generate some random values for each tip label in the data
d1 <- data.frame(id=tree2$tip.label, val=rnorm(30, sd=3))
# Make a second plot with the original, naming the new plot "dot", 
# using the data you just created, with a point geom.
p2 <- facet_plot(p, panel="dot", data=d1, geom=geom_point, aes(x=val), color='red3')
# Make some more data with another random value.
d2 <- data.frame(id=tree2$tip.label, value = abs(rnorm(30, mean=100, sd=50)))
# Now add to that second plot, this time using the new d2 data above, 
# This time showing a bar segment, size 3, colored blue.
p3 <- facet_plot(p2, panel='bar', data=d2, geom=geom_segment, 
           aes(x=0, xend=value, y=y, yend=y), size=3, color='blue4') 
# Show all three plots with a scale
p3 + theme_tree2()
```

## Overlay Organism Silouhettes

```{r}
# get phylopic 

newick <- "((Pongo_abelii,(Gorilla_gorilla_gorilla,(Pan_paniscus,Pan_troglodytes)Pan,Homo_sapiens)Homininae)Hominidae,Nomascus_leucogenys)Hominoidea;"

tree3 <- read.tree(text=newick)

d <- ggimage::phylopic_uid(tree3$tip.label)
d$body_mass = c(52, 114, 47, 45, 58, 6)

p <- ggtree(tree3) %<+% d + 
  geom_tiplab(aes(image=uid, colour=body_mass), geom="phylopic", offset=2.5) +
  geom_tiplab(aes(label=label), offset = .2) + xlim(NA, 7) +
  scale_color_viridis_c()

p
```
# Tips

If you the tree is not fitting in the plot space you can modify the xlim

```{r}
# Add to your 
ggtree(tree) +
  xlim(0,2) # This worked from my rectangular trees
  xlim(0,36) # This worked for my circular trees
```


To show the bootstrap values

```{r}
ggtree(tree) +  
  geom_nodelab()
```


To change text size and position

```{r}
ggtree(tree) +
  geom_tiplab(hjust=-.1, size = 3)
```

To change the size of the highlighted clade you can use extend

```{r}
ggtree(tree) +
  geom_hilight(node=17, fill="gold", extend = 0.5) 
```

You can order the layers when highlighting for certain effects



# Exercises

## Exercise 1

Use layout options in ggtree to provide a slanted phylogenetic tree, a circular tree, and a circular unscaled cladogram with thick red lines.

### Slanted Tree

```{r}
ggtree(tree, layout = 'slanted') + theme_tree2()
```

### Circular

```{r}
ggtree(tree, layout = 'circular') + theme_tree2()
```

### Circular Unscaled Cladogram with Thick Red Lines

```{r}
ggtree(tree, branch.length = "none", color = "red", size = 4, layout = 'circular') + theme_tree()
```

## Exercise 2

Create a phylogeny with the following aesthetic c haracteristics: tips labeled in purple, pruple colored diamond shape tip points, large semitransparent yellow node points, and add a title.

```{r}
ggtree(tree) +
  geom_tippoint(color = "purple", shape = "diamond") +
  geom_tiplab(color = "purple", size = 4) +
  geom_nodepoint(size = 4, color = "yellow", alpha = 0.5) +
  ggtitle("Creating a Colorful Phylogeny")
```

## Exercise 3

### Find what MRCA is for taxa B+C and L+J.

```{r}
MRCA(tree, c("B", "C"))

MRCA(tree, c("L", "J"))
```

### Draw the tree

Add tip labels, highlight the clades you found earlier, add a clade label to the larger superclade (node 17), Link taxa C to E, G to J, with a dashed grey line, add a scale bar, add a title

```{r}
ggtree(tree) +
  geom_tiplab() +
   geom_hilight(node=19, fill="gold") + 
  geom_hilight(node=23, fill="purple") +
  geom_cladelabel(node=17, label="Superclade 17", offset = 0.1, offset.text = 0.1, color="red2", align=TRUE) +
    theme_tree2() + 
  xlim(0, 75) +
  geom_taxalink("C", "E", color="grey", linetype = 2) +
  geom_taxalink("G", "J", color="grey", linetype = 2, curvature=-.9) +
  ggtitle("Fully Annotated Tree")
```

